humility-core/src/core/probe.rs

```diff
@@ -146,7 +144,7 @@ impl Core for ProbeCore {
         use num_traits::ToPrimitive;
 
         core.write_core_reg(
-            Into::<probe_rs::CoreRegisterAddress>::into(reg_id),
+            Into::<probe_rs::RegisterId>::into(reg_id),
             value as u32,
```

Comment:
We should only cast to a u32 on a 32bit platform.  I think this function is templated, so we can call a different version depending on our architecture.